pkg/placement/placement.go

```diff
@@ -179,40 +224,46 @@ func convertToClientObject(obj runtime.Object) (client.Object, error) {
 	return clientObj, nil
 }
 
-func updatePlacement(client dynamic.Interface, obj runtime.Object) error {
+func updatePlacement(client dynamic.Interface, placement *v1alpha1.Placement) error {
 	gvr := schema.GroupVersionResource{
 		Group:    v1alpha1.GroupVersion.Group,
-		Version:  obj.GetObjectKind().GroupVersionKind().Version,
-		Resource: "placements",
+		Version:  v1alpha1.GroupVersion.Version,
+		Resource: util.PlacementResource,
 	}
-	unstructuredObj, ok := obj.(*unstructured.Unstructured)
-	if !ok {
-		return fmt.Errorf("unexpected type for obj, expected *unstructured.Unstructured")
+
+	innerObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(placement)
+	if err != nil {
+		return fmt.Errorf("failed to convert Placement to unstructured: %w", err)
 	}
+
+	unstructuredObj := &unstructured.Unstructured{
+		Object: innerObj,
+	}
+
 	client.Resource(gvr).Namespace("").Update(context.Background(), unstructuredObj, metav1.UpdateOptions{})
 	return nil
 }
 
-func (c *Controller) deleteExternalResources(obj runtime.Object) error {
-	list, err := listManifestsForPlacement(c.ocmClient, c.wdsName, obj)
+func (c *Controller) deleteExternalResources(placement *v1alpha1.Placement) error {
+	list, err := listManifestsForPlacement(c.ocmClient, c.wdsName, placement)
 	if err != nil {
 		return err
 	}
-	mObj := obj.(metav1.Object)
-	labelKey := util.GenerateManagedByPlacementLabelKey(c.wdsName, mObj.GetName())
+
+	labelKey := util.GenerateManagedByPlacementLabelKey(c.wdsName, placement.GetName())
 	for _, manifest := range list.Items {
-		c.logger.Info("Trying to delete manifest", "manifest name", manifest.Name, "namespace", manifest.Namespace, "for placement", mObj.GetName())
+		c.logger.Info("Trying to delete manifest", "manifest name", manifest.Name,
```

Comment:
This log message is misleading, there is not necessarily an attempt to delete the ManifestWork here.

Reply:
This, along with object-wrapping/unwrapping code, will later be deleted as part of finalizing this work.